apollo-federation/src/query_plan/mod.rs

```diff
@@ -18,7 +18,7 @@ mod optimize;
 pub mod query_planner;
 pub(crate) mod query_planning_traversal;
 
-pub type QueryPlanCost = i64;
+pub type QueryPlanCost = f64;
```

Comment:
(by the way, rust also has an i/u128)

Reply:
The calculation is really meant to be floating point. One can craft a query that can blow out u128 by making a long `require` chain. 10-require chain breaks i64. 20 chain will surely break i128. Though it's unlikely in practice, it's a security issue.

Reply:
Alternatively, we can add a preventive logic. But, f64 is an easy solution and compatible with JS version.